cloudera-on-centos/azuredeploy.json

```diff
@@ -1,314 +1,407 @@
-            } 
-        }, 
-        "tshirtSize": { 
-            "type": "string", 
-            "defaultValue": "Eval", 
-            "allowedValues": [
-                "Eval",
-                "Prod"
-            ], 
-            "metadata": { 
-                "description": "T-shirt size of the Cloudera cluster (Eval, Prod)" 
-            } 
-        }, 
-        "numberOfDataNodes": {
-            "type": "int",
-            "defaultValue": 3,
-            "metadata": { 
-                "description": "Number of data nodes for Prod, default to 3" 
-            } 
-        },
-        "vmSize": {
-            "type": "string",
-            "defaultValue": "Standard_D14",
-            "metadata": { 
-                "description": "The size of the VMs deployed in the cluster (Defaults to Standard_D14)" 
-            } 
-        },
-        "keyVaultResourceGroup": {
-            "type": "string",
-            "defaultValue": "AzureRM-Util",
-            "metadata": { 
-                "description": "The resource group containing the key vault which provides the private key used for SSH login.  See the readme for details." 
-            } 
-        },
-        "keyVaultName": {
-            "type": "string",
-            "defaultValue": "AzureRM-Keys",
-            "metadata": { 
-                "description": "The name of the key vault which provides the private key used for SSH login.  See the readme for details." 
-            } 
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUserName": {
+      "type": "string",
+      "defaultValue": "testuser",
+      "metadata": {
+        "description": "Admin user name for the Virtual Machines"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "defaultValue": "Eur32#1e",
+      "metadata": {
+        "description": "Admin password for the Virtual Machines"
+      }
+    },
+    "cmUsername": {
+      "type": "string",
+      "defaultValue": "cmadmin",
+      "metadata": {
+        "description": "User name for the Cloudera Manager"
+      }
+    },
+    "cmPassword": {
+      "type": "securestring",
+      "defaultValue": "cmpassword",
+      "metadata": {
+        "description": "password for the Cloudera Manager"
+      }
+    },
+    "storageAccountPrefix": {
+      "type": "string",
+      "defaultValue": "storageaccountprefix",
+      "metadata": {
+        "description": "Unique namespace for the Storage Account where the Virtual Machine's disks will be placed"
+      }
+    },
+    "dnsNamePrefix": {
+      "type": "string",
+      "defaultValue": "dnsnameprefix",
+      "metadata": {
+        "description": "Unique public dns name where the Virtual Machines will be exposed"
+      }
+    },
+    "region": {
+      "type": "string",
+      "defaultValue": "japanwest",
```

Comment:
Please remove the default location and let the user pick from a set of allowed values